docs/sdg/sdg-refactor.md

```diff
@@ -0,0 +1,159 @@
+
+Conclusion:
+
+- The cost of having a separate repository is so high that we would only consider this option as a last resort.
+
+### Option 3: Move preprocessing and postprocessing into the instructlab/instructlab repository
+
+Pro:
+
+- The `instructlab/instructlab` repository already has a lot of "supporting" (non-core) functionality. It contains most user facing logic aside from what we call the "core" parts of the workflow (SDG, Train, Eval).  Since the preprocessing and postprocessing are non-code parts of SDG, this change would respect established precedent.  Examples of existing functionality that follow this pattern include all of the following and more:
+  - download
+  - serve
+  - chat
+  - list
+  - edit
+  - init
+- Supporting user flow 1.3.2 requires separate commands for preprocessing and core SDG.  This is slightly simpler if preprocessing is implemented in the `instructlab/instructlab` repository.  If preprocessing remains in the SDG library instead then the code in the `instructlab/instructlab` repository would need to make separate calls to the SDG library for preprocessing and core SDG to support user flow 1.3.2.  That adds a little complexity.
+- Avoids some of the cons of Option 1, but see below for some overlap.
+- Avoids some of the cons of Option 2, but see below for some overlap.
+
+Con:
+
+- Avoids the pros of both Option 1 and Option 2.
+- As with Option 1, this approach involves a lot of coordination.  There are a lot of stakeholders involved in the `instructlab/instructlab` repository and locating preprocessing and postprocessing there drags those stakeholders into issues relating to preprocessing and postprocessing.  However, as with Option 1, coordinating across stakeholders is something an open source project needs to do well anyway to remain engaged with the community.
+- As with Option 1, this approach suffers from the fact that keeping interconnected components in the same repository provides less pressure to consistently document the API contracts between them.  In the case of Option 1, the interconnected components that would not have as much pressure to be documented would be preprocessing/postprocessing and core SDG.  In the case of Option 3, the interconnected components that would not have as much pressure to be documented would be the command-line interface code and preprocessing/postprocessing.  However, in both cases, this con could be alleviated by just having the discipline to document the APIs well even without such pressure.
+- As with Option 2, this approach would not enable user flow 2.1.  That's fine since it is not on our requirements list.
+
+Conclusion:
+
+- This seems like a reasonable option.  The cons are mostly manageable.  However, overall the pros of Option 1 seem more compelling.
```

Comment:
This line might be better updated to reflect what is stated below under '## Decisions'

Reply:
Done!